runtime/reconcilers/util.go

```diff
@@ -154,3 +155,48 @@ func safeSQLName(name string) string {
 	}
 	return fmt.Sprintf("\"%s\"", strings.ReplaceAll(name, "\"", "\"\""))
 }
+
+func resolveTemplateProps(ctx context.Context, props map[string]any, self compilerv1.TemplateResource, c *runtime.Controller) (map[string]any, error) {
```

Comment:
The name sounds like it makes props for templating. Would also move contextual args to earlier in the arg list. Suggestion: `resolveTemplatedProps(ctx, c, self, props)`